core/runtime/src/main/java/io/quarkus/runtime/LaunchMode.java

```diff
@@ -23,12 +23,20 @@ public boolean isDevOrTest() {
     }
 
     /**
-     * Returns true if the current launch is the server side of remote dev.
+     * @return {@code true} if the current launch is the server side of remote dev.
      */
     public static boolean isRemoteDev() {
         return (current() == DEVELOPMENT) && "true".equals(System.getenv("QUARKUS_LAUNCH_DEVMODE"));
     }
 
+    /**
+     * @return {@code true} if the application is running in a WSL (Windows Subsystem for Linux) environment
+     */
+    public static boolean isWSL() {
+        var sysEnv = System.getenv();
+        return sysEnv.containsKey("IS_WSL") || sysEnv.containsKey("WSL_DISTRO_NAME");
+    }
+
```

Comment:
Let's not move that here. I don't think it's the right place and we need an implementation in SmallRye Common. We can duplicate this for now and we will drop both occurrences once we have the feature in SmallRye Common.